test/integration/controlplane/transformation/transformation_test.go

```diff
@@ -115,7 +115,9 @@ func newTransformTest(tb testing.TB, transformerConfigYAML string, reload bool,
 		return nil, fmt.Errorf("failed to read config file: %w", err)
 	}
 
-	if e.kubeAPIServer, err = kubeapiservertesting.StartTestServer(tb, nil, e.getEncryptionOptions(reload), e.storageConfig); err != nil {
+	if e.kubeAPIServer, err = kubeapiservertesting.StartTestServer(
+		tb, &kubeapiservertesting.TestServerInstanceOptions{BinaryVersion: "0.0"},
```

Comment:
this looks like it's causing failures in https://github.com/kubernetes/kubernetes/pull/125571/files#r1663216064

0.0 here is breaking CEL env initialization because required libraries declare their first available version as "1.0" (to indicate they are available in all versions with their feature)

changing this to 1.0 fails to even start the test server because APIs required to start and run the test (APIService, CRD, etc) don't even exist at that version.

Putting anything earlier than 1.10 (when APIService was added), 1.16 (when v1 CRD was added), or 1.30 (when VAP v1 was added) seems like it should solidly fail this test ... how did 0.0 work?

Reply:
There is a special logic in `deleted_kinds.go` to ignore version introduced when the version is 0.0
https://github.com/kubernetes/kubernetes/blob/f00eee5a40dfd46ad2092df56f7a9f514f3d3997/staging/src/k8s.io/apiserver/pkg/server/deleted_kinds.go#L103

Reply:
hrmm… why do we need to enable deleted kinds for this test?

Reply:
using a special case marker for one part of the system really trips up other parts of the system like cel registration

Reply:
`GetEtcdStorageDataForNamespace` is needed to ensure all non-alpha apis are stored correctly. But there is no single version that all the APIs exists for that version. We kind of had to hack it to cover all old/new/future APIs